### DISCUSSION TITLE
Script help

### DISCUSSION THREAD

**Discussion by razin1** [weather-gov/api#786] [Q&A]
Hi can someone help with this script so I can get weather alerts for the states mentioned...I keep getting console error, 0 weather alert fetched.  Not a computer person, so please be gentle with me :)
[Script.txt](https://github.com/user-attachments/files/18115876/Script.txt)

**Comment by dwhitemv25** [comment]
geometry is null for alerts unless the alert specifies a polygon. Otherwise the alert applies to the zones specified in the 'geocode' property.

If you want to plot a marker you'll need to determine a point in the zone to use. That, or instead of plotting a point, plot a polygon of the zone. The zone polygon can be fetched through the `/zones` endpoint.

This document gives a primer on translating alerts to locations:
https://www.weather.gov/media/documentation/docs/NWS_Geolocation.pdf

If you're trying to duplicate the alert map, you'll also need to sort the alerts by priority so the most important alert displays on top. The priority list can be found at https://www.weather.gov/help-map/.

**Comment by sullynole** [comment]
Also, there are services for alerts display if preferred:

https://opengeo.ncep.noaa.gov
https://www.weather.gov/gis/cloudgiswebservices

**Comment by razin1** [comment]
Thank you so much!

Sent from my mobile device.  Please excuse all typo's, grammatical error,
and terseness.

**Comment by razin1** [comment]
Thank you!

Sent from my mobile device.  Please excuse all typo's, grammatical error,
and terseness.

**Comment by razin1** [comment]
John quick question, how do I get the url template for the tiles for this
website - NWS Radar <https://radar.weather.gov/>.  Per your suggestion, it
might just be easier to have this as a layer for the map.  Thanks in
advance for all your help!

Also I tried to inspect, but could not get the url, thanks

**Reply by sullynole** [reply]
It is in the js source code - it's using the opengeo layers.  Can see the request in developer tools.  Example: https://opengeo.ncep.noaa.gov/geoserver/wwa/hazards/ows?REQUEST=GetMap&SERVICE=WMS&VERSION=1.1.1&FORMAT=image%2Fpng&TRANSPARENT=true&TILES=true&LAYERS=hazards&TIME=2024-12-13T19%3A38%3A20.000Z&WIDTH=512&HEIGHT=512&SRS=EPSG%3A3857&BBOX=-11897270.578531113%2C5009377.085697312%2C-11271098.442818949%2C5635549.221409475

**Comment by razin1** [comment]
thank you!
